packages/block-editor/src/components/block-list-appender/index.js

```diff
@@ -18,48 +18,72 @@ import IgnoreNestedEvents from '../ignore-nested-events';
 import DefaultBlockAppender from '../default-block-appender';
 import Inserter from '../inserter';
 
-function BlockListAppender( {
-	blockClientIds,
-	rootClientId,
-	canInsertDefaultBlock,
-	isLocked,
-} ) {
-	if ( isLocked ) {
-		return null;
-	}
-
-	if ( canInsertDefaultBlock ) {
-		return (
-			<IgnoreNestedEvents childHandledEvents={ [ 'onFocus', 'onClick', 'onKeyDown' ] }>
-				<DefaultBlockAppender
-					rootClientId={ rootClientId }
-					lastBlockClientId={ last( blockClientIds ) }
-				/>
-			</IgnoreNestedEvents>
-		);
-	}
+function renderDefaultAppender( rootClientId, lastBlockClientId ) {
+	return (
+		<IgnoreNestedEvents childHandledEvents={ [ 'onFocus', 'onClick', 'onKeyDown' ] }>
+			<DefaultBlockAppender
+				rootClientId={ rootClientId }
+				lastBlockClientId={ lastBlockClientId }
+			/>
+		</IgnoreNestedEvents>
+	);
+}
 
+function renderBlockAppender( rootClientId ) {
 	return (
 		<div className="block-list-appender">
 			<Inserter
 				rootClientId={ rootClientId }
 				renderToggle={ ( { onToggle, disabled, isOpen } ) => (
-					<IconButton
-						label={ __( 'Add block' ) }
-						icon="insert"
-						onClick={ onToggle }
+					<Button
 						className="block-list-appender__toggle"
-						aria-haspopup="true"
+						onClick={ onToggle }
 						aria-expanded={ isOpen }
 						disabled={ disabled }
-					/>
+					>
+						<Icon icon="insert" />
+						<span>{ __( 'Add Block' ) }</span>
+					</Button>
 				) }
 				isAppender
 			/>
 		</div>
 	);
 }
 
+function BlockListAppender( {
+	blockClientIds,
+	rootClientId,
+	canInsertDefaultBlock,
+	isLocked,
+	renderAppender,
+} ) {
+	if ( isLocked ) {
+		return null;
+	}
+
+	// If auto-insert Blocks is enabled, default to the standard behaviour
+	// of auto-inserting a Block but only if no renderAppender is provided
+	if ( ( ! renderAppender || renderAppender === 'auto-insert' ) && canInsertDefaultBlock ) {
+		return renderDefaultAppender( rootClientId, last( blockClientIds ) );
+	}
+
+	// Render prop - custom appender
+	if ( isFunction( renderAppender ) ) {
+		return (
+			<div className="block-list-appender">
```

Comment:
Maybe the components them selfs should add this class if they want to. For example div InnerBlocks.ButtonAppender includes so we are duplicating the div.

Reply:
I'm now only including this `<div>` when used within the `block-list-appender` file as this is appropriate. The raw included "appenders" don't include this class.